app/services/aspace_client.rb

```diff
@@ -5,20 +5,29 @@
 
 # Client for making requests via the ArchivesSpace API
 class AspaceClient
-  attr_reader :url
+  attr_reader :base_url
 
-  def initialize(url: ENV.fetch('ASPACE_URL', nil),
+  # rubocop:disable Metrics/MethodLength
+  def initialize(url: Settings.aspace.url,
                  user: ENV.fetch('ASPACE_USER', nil),
                  password: ENV.fetch('ASPACE_PASSWORD', nil))
-    unless url && user && password
-      raise ArgumentError,
-            'Please provide the url, user, and password for ArchivesSpace'
-    end
+    raise ArgumentError, 'Please provide the url for ArchivesSpace' unless url
 
-    @url = url
+    uri = URI.parse(url)
+    @base_url = url
     @user = user
     @password = password
+
+    return unless uri.user
+
+    @user ||= uri.user
+    @password ||= uri.password
+    @base_url = uri.dup.tap do |u|
+      u.user = nil
+      u.password = nil
+    end.to_s
```

Comment:
This addition doesn't seem related to the config gem changes.

Reply:
This was added in order to set the ASpace credentials more easily in development as a one line local config in settings.local.yml:

```
aspace:
  url: http://user:password@whatever.edu
```